lib/ex_aws/auth/presigned_posts/policy.ex

```diff
@@ -0,0 +1,34 @@
+defmodule ExAws.Auth.PresignedPosts.Policy do
+  @moduledoc false
+  import ExAws.Auth.Utils
+
+  defstruct [expiration: nil, conditions: []]
+
+  def new do
+    %__MODULE__{}
+  end
+
+  def set_expiration(policy, datetime) do
+    %__MODULE__{policy | expiration: datetime}
+  end
+
+  def add_condition(policy = %__MODULE__{conditions: conditions}, condition) do
+    new_conditions =
+      conditions
+      |> Enum.reverse
+      |> List.insert_at(0, condition)
+      |> Enum.reverse
+    %__MODULE__{policy | conditions: new_conditions}
```

Comment:
I sincerely appreciate the effort you put into to the docs and typespecs of these modules, but I feel like they obscure what's going on rather than actually helping to make it easier.